src/Parsers/ASTSelectQuery.cpp

```diff
@@ -474,4 +481,44 @@ void ASTSelectQuery::setFinal() // NOLINT method can be made const
     tables_element.table_expression->as<ASTTableExpression &>().final = true;
 }
 
+
+void ASTSelectQuery::setHasQueryParameters()
+{
+    if (!this->where())
+        return;
```

Comment:
In the general case it seems strange that `setHasQueryParameters` does not do anything in case `!this->where()`, because there are also parameters which are not related to parametrised view and it would have incorrect result.

Reply:
I removed this function to another visitor, and modified to check the whole query, not just where part.

Reply:
Sorry please ignore my above comment, I thought this was for getQueryParameters function which is now removed.

For checking query parameters, I have updated it to check whole query (not just where). Is it possible that query parameters can appear in other parts of the query?
[https://github.com/ClickHouse/ClickHouse/pull/41687#discussion_r980012738]
Also in ASTSelectWithUnionQuery, I am checking only inside all select statements `list_of_selects`, do you think it can appear anywhere else in this query too ?

Reply:
hm, yes, for example

```
CREATE VIEW test AS SELECT ... LIMIT {limit:UInt64}
```

so let's add tests for parameters in not only `WHERE`.

Reply:
added a test with `LIMIT {limit:UInt64}`